tests/test_sparql.py

```diff
@@ -50,7 +50,32 @@ def test_sparql(self):
                             for result in results))
         self.assertTrue(all(result['citizen_name'].toPython() == karl.name
                             for result in results))
-        results = sparql(query, session=None)
```

Comment:
The part above becomes almost redundant with the new checks. You can remove it (but do your test also for the three cases, `results_none`, `results_core_session` and `results_core_session_method`).